src/core/server/elasticsearch/client/configure_client.ts

```diff
@@ -35,6 +35,8 @@ export const configureClient = (
     ...clientOptions,
     Transport: KibanaTransport,
     Connection: HttpConnection,
+    // using ClusterConnectionPool until https://github.com/elastic/elasticsearch-js/issues/1714 is addressed
+    ConnectionPool: ClusterConnectionPool,
```

Comment:
I didn't add an integration test here because:
- Ihmo we shouldn't be testing the `@elastic/elasticsearch` behavior from Kibana tests
- The test would be quite long (starting multiple ES nodes at least twice), and by nature, likely flaky

If we really want, I can add one though